deepchem/models/torch_models/infograph.py

```diff
@@ -101,93 +293,169 @@ class InfoGraphModel(ModularTorchModel):
+                                         self.embedding_dim,
+                                         (self.embedding_dim,)),
+                'ff2':
+                    MultilayerPerceptron(self.embedding_dim, self.embedding_dim,
+                                         (self.embedding_dim,)),
+                'fc1':
+                    torch.nn.Linear(2 * self.embedding_dim, self.embedding_dim),
+                'fc2':
+                    torch.nn.Linear(self.embedding_dim, 1),
+                'local_d':
+                    MultilayerPerceptron(self.embedding_dim,
+                                         self.embedding_dim,
+                                         (self.embedding_dim,),
+                                         skip_connection=True),
+                'global_d':
+                    MultilayerPerceptron(self.embedding_dim,
+                                         self.embedding_dim,
+                                         (self.embedding_dim,),
+                                         skip_connection=True)
+            }
 
     def build_model(self):
         """
         Builds the InfoGraph model by unpacking the components dictionary and passing them to the InfoGraph nn.module.
         """
-        return InfoGraph(**self.components)
+        return InfoGraphStar(**self.components)
 
     def loss_func(self, inputs, labels, weights):
-        if self.use_unsup_loss:
+        if self.task == 'semisupervised':
             sup_loss = F.mse_loss(self.model(inputs), labels)
-            unsup_loss = self.unsup_loss(inputs)
-            if self.separate_encoder:
-                unsup_sup_loss = self.unsup_sup_loss(inputs, labels, weights)
-                loss = sup_loss + unsup_loss + unsup_sup_loss * self.learning_rate
-            else:
-                loss = sup_loss + unsup_loss * self.learning_rate
+            local_unsup_loss = self.local_unsup_loss(inputs)
+            global_unsup_loss = self.global_unsup_loss(inputs, labels, weights)
+            loss = sup_loss + local_unsup_loss + global_unsup_loss * self.learning_rate
+            # original implementation also includes an option if not using a separate encoder:
+            # loss = sup_loss + local_unsup_loss * self.learning_rate
             return (loss * weights).mean()
         else:
             sup_loss = F.mse_loss(self.model(inputs), labels)
             return (sup_loss * weights).mean()
 
-    def unsup_loss(self, inputs):
-        if self.separate_encoder:
+    def local_unsup_loss(self, inputs):
+        if self.task == 'semisupervised':
             y, M = self.components['unsup_encoder'](inputs)
         else:
             y, M = self.components['encoder'](inputs)
         g_enc = self.components['global_d'](y)
         l_enc = self.components['local_d'](M)
 
         if self.local:
-            loss = self._local_global_loss(l_enc, g_enc, inputs.graph_index)
+            loss = self.MIloss.local_global_loss(l_enc, g_enc,
```

Comment:
We should follow existing convention for how losses behave, see https://github.com/deepchem/deepchem/blob/master/deepchem/models/losses.py#L1 and how `TorchModel` uses losses. See also my comments above on the loss class itself